src/core/pointcloud/qgspointcloudrenderer.cpp

```diff
@@ -242,6 +244,7 @@ void QgsPointCloudRenderer::restoreCommonProperties( const QDomElement &element,
     mLabelTextFormat = QgsTextFormat();
     mLabelTextFormat.readXml( element.firstChildElement( QStringLiteral( "text-style" ) ), context );
   }
+  mZoomOutBehavior = static_cast<Qgis::PointCloudZoomOutRenderBehavior>( element.attribute( QStringLiteral( "zoomOutBehavior" ), QStringLiteral( "0" ) ).toInt() );
```

Comment:
A little nicer is to use qgsEnumValueToKey/qgsEnumKeyToValue for serialising enums instead

Reply:
I didn't know this function existed, I just reused the same approach that was already used in the code. Refactored!